x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/MonitoringTemplateRegistryTests.java

```diff
@@ -0,0 +1,432 @@
+                return AcknowledgedResponse.TRUE;
+            } else if (action instanceof PutComponentTemplateAction) {
+                // Ignore this, it's verified in another test
+                return new MonitoringTemplateRegistryTests.TestPutIndexTemplateResponse(true);
+            } else if (action instanceof PutComposableIndexTemplateAction) {
+                // Ignore this, it's verified in another test
+                return AcknowledgedResponse.TRUE;
+            } else if (action instanceof PutIndexTemplateAction) {
+                return AcknowledgedResponse.TRUE;
+            } else {
+                fail("client called with unexpected request: " + request.toString());
+                return null;
+            }
+        });
+
+        ClusterChangedEvent event = createClusterChangedEvent(Collections.emptyMap(), nodes);
+        if (historyDurationPresent) {
+            Settings testSettings = Settings.builder().put(HISTORY_DURATION.getKey(), expectedDeleteMinAge).build();
+            MonitoringTemplateRegistry testRegistry = new MonitoringTemplateRegistry(
+                testSettings,
+                clusterService,
+                threadPool,
+                client,
+                NamedXContentRegistry.EMPTY
+            );
+            testRegistry.clusterChanged(event);
+        } else {
+            registry.clusterChanged(event);
+        }
+        assertBusy(() -> assertThat(calledTimes.get(), equalTo(1)));
+
+        // Make sure we logged a deprecation for using the history setting
+        if (historyDurationPresent) {
+            assertSettingDeprecationsAndWarnings(new Setting<?>[]{HISTORY_DURATION});
+        }
+    }
+
+    public void testPolicyAlreadyExists() {
+        DiscoveryNode node = new DiscoveryNode("node", ESTestCase.buildNewFakeTransportAddress(), Version.CURRENT);
+        DiscoveryNodes nodes = DiscoveryNodes.builder().localNodeId("node").masterNodeId("node").add(node).build();
+
+        Map<String, LifecyclePolicy> policyMap = new HashMap<>();
+        List<LifecyclePolicy> policies = registry.getPolicyConfigs();
+        assertThat(policies, hasSize(1));
+        policies.forEach(p -> policyMap.put(p.getName(), p));
+
+        client.setVerifier((action, request, listener) -> {
+            if (action instanceof PutComponentTemplateAction) {
+                // Ignore this, it's verified in another test
+                return AcknowledgedResponse.TRUE;
+            } else if (action instanceof PutComposableIndexTemplateAction) {
+                // Ignore this, it's verified in another test
+                return AcknowledgedResponse.TRUE;
+            } else if (action instanceof PutIndexTemplateAction) {
+                // Ignore this, it's verified in another test
+                return AcknowledgedResponse.TRUE;
+            } else if (action instanceof PutLifecycleAction) {
+                fail("if the policy already exists it should be re-put");
```

Comment:
Should this be:
```suggestion
                fail("if the policy already exists it should not be re-put");
```
?

Reply:
Ahh yeah that seems to be an issue on the other test as well. I'll correct it in both places.